### ISSUE TITLE
cppcoreguidelines-narrowing-conversions doesn't trigger for templated function arguments

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#46242]
|  |  |
| --- | --- |
| Bugzilla Link | [46898](https://llvm.org/bz46898) |
| Version | unspecified |
| OS | All |
| Reporter | LLVM Bugzilla Contributor |
| CC | @legrosbuffle,@EugeneZelenko |

## Extended Description 
Reproducer:
void f1(float);
template<class T> void f2(T);
void t(double d) {
    f1(d);
    f2<float>(d);
}

The call to f1 and f2 are identical. In both cases passing the argument causes a narrowing conversion. They should both give a warning.

https://godbolt.org/z/cbnr47

**Comment by PiotrZSL** [comment]
According to gobolt both issues are detected, so there is no problem on main branch.

```
[<source>:4:8: warning: narrowing conversion from 'double' to 'float' [cppcoreguidelines-narrowing-conversions]](javascript:;)
    f1(d);
       ^
[<source>:5:15: warning: narrowing conversion from 'double' to 'float' [cppcoreguidelines-narrowing-conversions]](javascript:;)
    f2<float>(d);
```
